lib/iris/coords.py

```diff
@@ -52,6 +52,697 @@
+        if hasattr(self._values, 'to_xml_attr'):
+            element.setAttribute(values_term, self._values.to_xml_attr())
+        else:
+            element.setAttribute(values_term,
+                                 iris.util.format_array(self._values))
+        return element
+
+    def _xml_id_extra(self, unique_value):
+        return unique_value
+
+    def _xml_id(self):
+        # Returns a consistent, unique string identifier for this coordinate.
+        unique_value = b''
+        if self.standard_name:
+            unique_value += self.standard_name.encode('utf-8')
+        unique_value += b'\0'
+        if self.long_name:
+            unique_value += self.long_name.encode('utf-8')
+        unique_value += b'\0'
+        unique_value += str(self.units).encode('utf-8') + b'\0'
+        for k, v in sorted(self.attributes.items()):
+            unique_value += (str(k) + ':' + str(v)).encode('utf-8') + b'\0'
+        # Extra modifications to unique_value that are specialised in child
+        # classes
+        unique_value = self._xml_id_extra(unique_value)
+        # Mask to ensure consistency across Python versions & platforms.
+        crc = zlib.crc32(unique_value) & 0xffffffff
+        return '%08x' % (crc, )
+
+    def _value_type_name(self):
+        """
+        A simple, readable name for the data type of the dimensional metadata
+        values.
+
+        """
+        dtype = self._core_values().dtype
+        kind = dtype.kind
+        if kind in 'SU':
+            # Establish the basic type name for 'string' type data.
+            # N.B. this means "unicode" in Python3, and "str" in Python2.
+            value_type_name = 'string'
+
+            # Override this if not the 'native' string type.
+            if six.PY3:
+                if kind == 'S':
+                    value_type_name = 'bytes'
+            else:
+                if kind == 'U':
+                    value_type_name = 'unicode'
+        else:
+            value_type_name = dtype.name
+
+        return value_type_name
+
+
+class AncillaryVariable(_DimensionalMetadata):
+    def __init__(self, data, standard_name=None, long_name=None,
+                 var_name=None, units='no-unit', attributes=None):
+        """
+        Constructs a single ancillary variable.
+
+        Args:
+
+        * values:
+            The values of the ancillary variable.
+
+        Kwargs:
+
+        * standard_name:
+            CF standard name of the ancillary variable.
+        * long_name:
+            Descriptive name of the ancillary variable.
+        * var_name:
+            The netCDF variable name for the ancillary variable.
+        * units
+            The :class:`~cf_units.Unit` of the ancillary variable's values.
+            Can be a string, which will be converted to a Unit object.
+        * attributes
+            A dictionary containing other cf and user-defined attributes.
+
+        """
+        super(AncillaryVariable, self).__init__(
```

Comment:
An awful lot of your 'super' calls can be replaced by Python 3 argumentless ones, I think.
I.E. `super(SelfClass, self)` --> `super(self.__class__, self)` == `super()`.
Much nicer I think !
Applies a lot here, and in Coord.